iota-client/src/client.rs

```diff
@@ -292,21 +291,22 @@ impl Client {
     pub(crate) async fn sync_nodes(
         sync: &Arc<RwLock<HashSet<Url>>>,
         nodes: &HashSet<Url>,
-        local_pow: bool,
-        network: Network,
+        network_info: &Arc<RwLock<NetworkInfo>>,
     ) {
         let mut synced_nodes = HashSet::new();
 
         for node_url in nodes {
             // Put the healty node url into the synced_nodes
             if let Ok(info) = Client::get_node_info(node_url.clone()).await {
                 if info.is_healthy {
-                    if network == Network::Testnet && info.network_id == "mainnet"
-                        || network == Network::Mainnet && info.network_id != "mainnet"
+                    if network_info.read().unwrap().network == Network::Testnet && info.network_id == "mainnet"
+                        || network_info.read().unwrap().network == Network::Mainnet && info.network_id != "mainnet"
                     {
                         continue;
                     }
-                    if !local_pow {
+                    let mut client_network_info = network_info.write().unwrap();
+                    client_network_info.min_pow_score = info.min_pow_score;
```

Comment:
should we also set `network_info.local_pow` to true if the node doesn't have the PoW feature?

Reply:
No, then the node should not be selected

Reply:
At least that's what I think, because local_pow is true as default and if you disable it you should have a reason for it, maybe running on a device with a slow cpu, automatically enabling it could cause problems or make the message useless, if the PoW takes too long